tests/test_resources/test_clusters/conftest.py

```diff
@@ -343,9 +342,67 @@ def local_docker_cluster_public_key(request):
     )
     local_ssh_port = BASE_LOCAL_SSH_PORT + 2
 
+    client, rh_parent_path = build_and_run_image(
+        image_name=image_name,
+        container_name=container_name,
+        dir_name=dir_name,
+        detached=detached,
+        keypath=keypath,
+        force_rebuild=request.config.getoption("--force-rebuild"),
+        port_fwds=[f"{local_ssh_port}:22", "32300:32300"],
```

Comment:
When I ran this I got "RuntimeError: Failed to run docker image keypair-telemetry: docker: Error response from daemon: driver failed programming external connectivity on endpoint rh-slim-keypair-telemetry (f8b9d98d5c96cd3e812949aa8cc850817ec607ebf49b4240a67e1f8263a9c010): Bind for 0.0.0.0:32300 failed: port is already allocated." Another container was up so it may have been a port conflict. You may need to choose a different port on the local side for the mapping. Once I took down the other container the test passed!

Reply:
Good catch! I had the same error early on in testing, then started running only one test until it passed. 
I think Rohin's PR should fix this, happy to take another look if it isn't.

Reply:
@rohinb2 FYI